circuits/aes-gcm/aes-gcm.circom

```diff
@@ -0,0 +1,91 @@
+pragma circom 2.1.9;
+
+include "../aes-ctr/ctr.circom";
+include "../ghash/ghash.circom";
+include "../aes-ctr/cipher.circom";
+include "circomlib/circuits/bitify.circom";
+include "helper_functions.circom";
+
+
+/// AES-GCM with 128 bit key according to: https://nvlpubs.nist.gov/nistpubs/legacy/sp/nistspecialpublication800-38d.pdf
+/// 
+/// Parameters:
+/// l: length of the plaintext
+///
+/// Inputs:
+/// key: 128-bit key
+/// iv: initialization vector
+/// plainText: plaintext to be encrypted
+/// additionalData: additional data to be authenticated
+///
+/// Outputs:
+/// cipherText: encrypted ciphertext
+/// authTag: authentication tag
+/// 
+template AESGCM(l) {
+    // Inputs
+    signal input key[16]; // 128-bit key
+    signal input iv[12]; // IV length is 96 bits (12 bytes)
+    signal input plainText[l];
+    signal input additionalData[16]; // AAD length is 128 bits (16 bytes)
+
+    // Outputs
+    signal output cipherText[l];
+    signal output authTag[16]; // Authentication tag length is 128 bits (16 bytes)
```

Comment:
I believe in a folding configuration, we'd need to output something like an "intermediate auth tag" that can be used as input in the second chunk.  That also means we'd need a template which accepts the prior auth tag.